core/selection.js

```diff
@@ -1981,7 +1981,7 @@
 			}
 
 			// Handle special case - fake selection of table cells.
-			if ( editor && editor.config.tableImprovements && isTableSelection( ranges ) && !isSelectingTable ) {
+			if ( editor && editor.plugins.tableselection && isTableSelection( ranges ) && !isSelectingTable ) {
```

Comment:
Oh, yes, the problem here is that the plugin will get loaded despite table selection not being supported on a given environment. We need additional check to `CKEDITOR.plugins.tableselection.isSupportedEnvironment`.